### DISCUSSION TITLE
no case clause matching: {:plugin_module_unloadable, 'prometheus_process_collector'

### DISCUSSION THREAD

**Discussion by zjyang861** [rabbitmq/rabbitmq-server#5247] [Other] [answered]
Thank you for using RabbitMQ.

**STOP NOW AND READ THIS** BEFORE OPENING A NEW ISSUE ON GITHUB

Unless you are CERTAIN you have found a reproducible problem in RabbitMQ or
have a **specific, actionable** suggestion for our team, you must first ask
your question or discuss your suspected issue on the mailing list:

https://groups.google.com/forum/#!forum/rabbitmq-users

Team RabbitMQ does not use GitHub issues for discussions, investigations, root
cause analysis and so on.

Please take the time to read the CONTRIBUTING.md document for instructions on
how to effectively ask a question or report a suspected issue:

https://github.com/rabbitmq/rabbitmq-server/blob/master/CONTRIBUTING.md#github-issues

Following these rules **will save time** for both you and RabbitMQ's maintainers.

Thank you.

**Comment by zjyang861** [comment]
** (CaseClauseError) no case clause matching: {:plugin_module_unloadable, 'prometheus_process_collector', {:error, :on_load_failure}}
    (rabbitmqctl) lib/rabbitmq/cli/plugins/plugins_helpers.ex:218: RabbitMQ.CLI.Plugins.Helpers.update_enabled_plugins/2
    (rabbitmqctl) lib/rabbitmq/cli/plugins/plugins_helpers.ex:115: RabbitMQ.CLI.Plugins.Helpers.update_enabled_plugins/4
    (rabbitmqctl) lib/rabbitmq/cli/plugins/commands/enable_command.ex:104: anonymous fn/6 in RabbitMQ.CLI.Plugins.Commands.EnableCommand.do_run/2
    (elixir) lib/stream.ex:1298: anonymous fn/2 in Stream.iterate/2
    (elixir) lib/stream.ex:1482: Stream.do_unfold/4
    (elixir) lib/stream.ex:1553: Enumerable.Stream.do_each/4
    (elixir) lib/stream.ex:933: Stream.do_enum_transform/7
    (elixir) lib/stream.ex:1553: Enumerable.Stream.do_each/4

**Comment by zjyang861** [comment]
I am getting this error when I enable the plugin, what is the problem?

**Comment by michaelklishin** [comment]
We cannot suggest much without knowing what RabbitMQ and Erlang versions are used.

**Comment by michaelklishin** [comment, marked as answer]
Also, you can [pre-enabled plugins](https://rabbitmq.com/plugins.html#enabled-plugins-file) to compare.

My best guess is that you may have an old version of `rabbitmq_prometheus` (it used to be a community plugin) on the code path but try to use a modern version that ships with RabbitMQ on Erlang 25.

Erlang 25 won't be able to load modules compiled on Erlang 21 and older. That means most (if not all) older `rabbitmq_prometheus` releases.

**Comment by ikavgo** [comment]
Agree, to rebuild process collector, please clone it and run `rebar3 archive` - it will rebuild. then just copy the archive to usual plugins directory

**Comment by ikavgo** [comment]
BTW why you didn't migrate to built-in plugin? what's missing?

**Reply by michaelklishin** [reply]
Most likely the old plugin simply wasn't removed from the code path (e.g. the image deployed).
